core/services/bridget/settings.py

```diff
@@ -42,3 +42,58 @@ def migrate(self, data: Dict[str, Any]) -> None:
             super().migrate(data)
 
         data["VERSION"] = SettingsV1.VERSION
+
+
+class BridgeSettingsSpecV2(pykson.JsonObject):
+    udp_target_port = pykson.IntegerField()
+    udp_listen_port = pykson.IntegerField()
+    serial_path = pykson.StringField()
+    baudrate = pykson.IntegerField()
+    ip = pykson.StringField()
+
+    @staticmethod
+    def from_spec(spec: "BridgeFrontendSpec") -> "BridgeSettingsSpecV2":  # type: ignore
+        return BridgeSettingsSpecV2(
+            serial_path=spec.serial_path,
+            baudrate=spec.baud,
+            ip=spec.ip,
+            udp_target_port=spec.udp_target_port,
+            udp_listen_port=spec.udp_listen_port,
+        )
+
+    def __eq__(self, other: object) -> Any:
+        if isinstance(other, BridgeSettingsSpecV2):
+            return self.serial_path == other.serial_path
+        return False
+
+
+class SettingsV2(SettingsV1):
+    VERSION = 2
+    specsv2 = pykson.ObjectListField(BridgeSettingsSpecV2)
+
+    def __init__(self, *args: str, **kwargs: int) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.VERSION = SettingsV2.VERSION
+
+    def migrate(self, data: Dict[str, Any]) -> None:
+        if data["VERSION"] == SettingsV2.VERSION:
+            return
+
+        if data["VERSION"] < SettingsV2.VERSION:
+            super().migrate(data)
+
+            data["VERSION"] = SettingsV2.VERSION
+            new_specs = []
+            for spec in data["specs"]:
+                server = spec["ip"] == "0.0.0.0"
+                new_specs.append(
+                    {
+                        "serial_path": spec["serial_path"],
+                        "baudrate": spec["baudrate"],
+                        "ip": spec["ip"],
+                        "udp_target_port": 0 if server else spec["udp_port"],
+                        "udp_listen_port": spec["udp_port"] if server else 0,
+                    }
+                )
+            data["specsv2"] = new_specs
```

Comment:
why not just replacing specs directly ?

Reply:
sounds good, that is just how my brain works. updated.